d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -306,20 +300,20 @@ public void onReconnect()
     };
   }
 
-  XdsClient.D2NodeMapResourceWatcher getUriResourceWatcher(String clusterName)
+  XdsClient.NodeMapResourceWatcher getUriResourceWatcher(String clusterName)
   {
     return new UriPropertiesResourceWatcher(clusterName);
   }
 
-  XdsClient.D2SymlinkNodeResourceWatcher getSymlinkResourceWatcher(String symlinkName)
+  XdsClient.SymlinkNodeResourceWatcher getSymlinkResourceWatcher(String symlinkName)
   {
-    return new XdsClient.D2SymlinkNodeResourceWatcher()
+    return new XdsClient.SymlinkNodeResourceWatcher()
     {
       @Override
-      public void onChanged(String resourceName, XdsClient.D2SymlinkNodeUpdate update)
+      public void onChanged(String resourceName, XdsClient.NodeUpdate update)
       {
         // Update maps between symlink name and actual node name
-        String actualResourceName = update.getNodeData().getMasterClusterNodePath();
+        String actualResourceName = update.getNodeData().getData().toString(StandardCharsets.UTF_8);
```

Comment:
Won't this get the `bytes data = 2` field from `Node`, isn't that the whole node data and not just the resource name?

Or if there is some semantic meaning to `data` (like it's bytes for only the resource name) we should probably document that, either here or in the proto

Reply:
This has been updated, but for the record the node's data _is_ the symlink. Symlink nodes don't have JSON in them, they just have a string